subprojects/core/src/main/java/org/gradle/api/internal/tasks/cache/diagnostics/TaskOutputCacheDiagnostics.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.cache.diagnostics;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskOutputCacheDiagnostics {
+    List<TaskOutputCacheEvent> events = new ArrayList<TaskOutputCacheEvent>();
+
+    public void event(TaskOutputCacheEvent event) {
+        events.add(event);
+    }
+
+    public int getExecutedTasks() {
+        return events.size();
+    }
+
+    public int getCachedTasks() {
+        int cached = 0;
+        for (TaskOutputCacheEvent event : events) {
+            if (event.isCached()) {
+                cached++;
+            }
+        }
+        return cached;
+    }
+
+    public int getCacheableTasks() {
+        int cacheable = 0;
+        for (TaskOutputCacheEvent event : events) {
+            if (event.isCacheable()) {
+                cacheable++;
+            }
+        }
+        return cacheable;
+    }
```

Comment:
The names `get*Tasks()` are a bit misleading, as the methods don't return the tasks themselves but count them instead. I think the names of the methods should reflect that.